src/app/article-list/article-list.component.ts

```diff
@@ -1,14 +1,23 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ChangeDetectorRef, AfterViewInit  } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, NavigationError, NavigationStart, Router, Event } from '@angular/router';
+import { ArticleListService } from '../services/article-list.service';
 
 @Component({
   selector: 'app-article-list',
   templateUrl: './article-list.component.html',
-  styleUrls: ['./article-list.component.css']
+  styleUrls: ['./article-list.component.css'],
+  providers: [ArticleListService]
 })
 export class ArticleListComponent implements OnInit {
 
+
+
   isActive = false;
-  postzSorted: any;
+  articlesList: any;
```

Comment:
unikamy stosowania `any` jesli wiemy jaki jest typ podajemy, jesli typ moze sie roznic to uzywamy `unknown` lub `never` w zaleznosci od potrzeb